File: TreasurersApp/Models/PhoneNumberActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class PhoneNumberActionResult : WebResult<PhoneNumber>
    {
        public PhoneNumberActionResult(bool success, List<string> statusMessages, PhoneNumber phoneNumber)
            :base(success, statusMessages, phoneNumber)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/PhoneNumberActionResult.ts
import { PhoneNumber } from './PhoneNumber';
import { PhoneService } from '../maintenance/phone-maintenance/phone.service';

export class PhoneNumberActionResult {
  success = false;
  statusMessages: string[] = [];
  phoneNumber: PhoneNumber = null;

  constructor(private phoneService: PhoneService) {
    this.success = false;
    this.phoneNumber = this.phoneService.newPhoneNumber();
    this.statusMessages = [];
  }
}
